libs/expo/betterangels/src/lib/screens/AddEditClient/PersonalInfo/FullName.tsx

```diff
@@ -1,26 +1,58 @@
 import { Colors } from '@monorepo/expo/shared/static';
-import { CardWrapper, Input } from '@monorepo/expo/shared/ui-components';
-import { useFormContext } from 'react-hook-form';
+import {
+  CardWrapper,
+  Input,
+  TextRegular,
+} from '@monorepo/expo/shared/ui-components';
+import { useEffect } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
 import { Text } from 'react-native';
 import {
   CreateClientProfileInput,
   UpdateClientProfileInput,
 } from '../../../apollo';
 
+type TFullNameForm = (UpdateClientProfileInput | CreateClientProfileInput) & {
+  emptyFormError: string;
+};
+
 export default function FullName() {
   const {
     control,
     formState: { errors },
+    setError,
     setValue,
-  } = useFormContext<UpdateClientProfileInput | CreateClientProfileInput>();
+    clearErrors,
+  } = useFormContext<TFullNameForm>();
 
   const onReset = () => {
     setValue('user.firstName', '');
     setValue('user.middleName', '');
     setValue('user.lastName', '');
     setValue('nickname', '');
+    clearErrors('emptyFormError');
   };
 
+  const [firstName, middleName, lastName, nickname] = useWatch({
+    control,
+    name: ['user.firstName', 'user.middleName', 'user.lastName', 'nickname'],
+  });
+
+  useEffect(() => {
+    const hasValue = firstName || middleName || lastName || nickname;
+
+    if (hasValue) {
+      clearErrors('emptyFormError');
+
+      return;
+    }
+
+    setError('emptyFormError', {
+      type: 'manual',
+      message: 'Filling out one of the fields is required',
+    });
+  }, [firstName, middleName, lastName, nickname, setError, clearErrors]);
```

Comment:
why not input level error check rather than in use effect? I would say just create a rule required if other values are not there

Reply:
I looked into it and still still having some issues, so let's reconnect tomorrow on this. 
But here are my thoughts:

1. I couldn't make it work properly, so that's the big one lol. I agree with you that this approach should work, so maybe we can pair tomorrow?
2. `Filling out one of the fields is required` scenario  is not super common, so seems ok to treat it as a one-off, but it still feels a bit strange to put a `required` rule on an input when that specific input isn't really "required" (only any 1 out of 4 is required). Not huge of an issue since it's a one-off, but 🤷